### PR TITLE
Separate affine_channel from math and optimize it

### PR DESCRIPTION
**Pull request by xiaomengy**
Summary:
Separate affine_channel from math and optimize it

i-am-not-moving-c2-to-c10

Differential Revision: D13727606

### CODE REVIEW

**Comment by BIT-silence** [comment]
Benchmark:
N = 256, C = 256, H = 128, W = 128

AffineChannelOp NCHW order: 67.47 -> 38.32ms
AffineChannelOp NHWC order: 103.95-> 39.49ms

SpatialBN NCHW testing: 38.31ms, CuDNN 43.31ms
SpatialBN NCHW training: 69.06ms, CuDNN 66.58ms

File: caffe2/utils/math/elementwise_level2.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#ifndef CAFFE2_UTILS_MATH_ELEMENTWISE_H_
+#define CAFFE2_UTILS_MATH_ELEMENTWISE_H_
+
```
Comment (houseroad) [inline caffe2/utils/math/elementwise_level2.h]: Can we have a more meaningful name for this header/cc file? And add some comments talking about which functions should be put in this file?

Comment (BIT-silence) [reply caffe2/utils/math/elementwise_level2.h]: The AffineChannel is a special case of elementwise mul and add operation, so I think it's fine to make it in elementwise.h

**Comment by houseroad** [comment]
The change breaks the hip build. @bddppq any suggestion?

**Review by houseroad — APPROVED** [review]
LGTM

**Pull request closed without merging**